test/modular/TraceurTestRunner.js

```diff
@@ -38,70 +28,34 @@ export class TraceurTestRunner extends Mocha6 {
     return this.defaultOptions_;
   }
 
-  // Reads process arguments, merges defaults options, returns options object.
-  parseCommandLine() {
-    let testOptions = this.defaultOptions();
-    let commandLine =
-        new Command(process.argv[0] + ' ' + process.argv[1]);
-
-    Object.getOwnPropertyNames(testOptions).forEach((prop) => {
-      commandLine[prop] = testOptions[prop];
-    });
-
-    commandLine.option('-?, --help', 'Show this help text', () => {
-      commandLine.help();
-    }).usage('[options] [files]')
-
-    // Selected mocha options supported.
-    commandLine.option('-g, --grep <pattern>', 'only run tests matching <pattern>').
-        option('-i, --invert', 'inverts --grep matches').
-        option('-b, --bail', 'bail after first test failure');
-
-    commandLine.command('only <file> [files...]').
-        description('only test these [files] ').action(
-        (file, files) => {
-          commandLine.patterns = [file];
-          if (files)
-            commandLine.patterns = commandLine.patterns.concat(files);
-        });
-
-    commandLine.parse(process.argv);
-
-    return commandLine;
-  }
-
-  applyOptions(testOptions) {
+  applyOptions(patterns) {
     // Apply the mocha options
+    var testOptions = this.getOptions();
     if (testOptions.grep)
       this.grep(new RegExp(testOptions.grep));
     if (testOptions.invert)
       this.invert();
     if (testOptions.bail)
       this.bail();
-
-    testOptions.patterns.forEach((pattern) => {
-      let files = glob.sync(pattern, {});
-      files.forEach((file) => this.addFile(file));
-    });
+    this.patterns_ = patterns;
   }
 
   run() {
     let failed = 0;
-    super.run().then((runner) => {
-      runner.on('fail', (err) => {
-        failed++;
-      });
-      runner.on('end', () => {
-        process.exit(failed);
+    return this.expandPatterns().then(() => {
+      return super.run().then((runner) => {
+        return new Promise((resolve, reject) => {
+          runner.on('fail', (err) => {
+            failed++;
+          });
+          runner.on('end', () => {
+            resolve(failed);
```

Comment:
No, we are counting failures and resolving to the number (for the old school return code).  I changed the var name to number of failures.